RNCAsyncStorage.podspec

```diff
@@ -10,10 +12,30 @@ Pod::Spec.new do |s|
 
   s.authors      = package['author']
   s.homepage     = package['homepage']
-  s.platforms    = { :ios => "9.0", :tvos => "9.2", :osx => "10.14" }
 
   s.source       = { :git => "https://github.com/react-native-async-storage/async-storage.git", :tag => "v#{s.version}" }
-  s.source_files  = "ios/**/*.{h,m}"
+  s.source_files  = "ios/**/*.{h,m,mm}"
+
+  if fabric_enabled
+    folly_compiler_flags = '-DFOLLY_NO_CONFIG -DFOLLY_MOBILE=1 -DFOLLY_USE_LIBCPP=1 -Wno-comma -Wno-shorten-64-to-32'
+
+    s.pod_target_xcconfig = {
+      'HEADER_SEARCH_PATHS' => '"$(PODS_ROOT)/boost" "$(PODS_ROOT)/boost-for-react-native" "$(PODS_ROOT)/RCT-Folly"',
+      'CLANG_CXX_LANGUAGE_STANDARD' => 'c++17',
+    }
+    s.platforms       = { ios: '12.4', tvos: '11.0', :osx => "10.15" }
+    s.compiler_flags  = folly_compiler_flags + ' -DRCT_NEW_ARCH_ENABLED'
+
+    s.dependency "React"
+    s.dependency "React-RCTFabric" # This is for fabric component
+    s.dependency "React-Codegen"
+    s.dependency "RCT-Folly"
+    s.dependency "RCTRequired"
+    s.dependency "RCTTypeSafety"
+    s.dependency "ReactCommon/turbomodule/core"
```

Comment:
Are all of these used in the generated `rnasyncstorage.h`? Otherwise, I can't see them being used in this PR.

Reply:
Since we don't want to support below 0.71, I've replaced those with [install_modules_dependencies](https://reactnative.dev/docs/new-architecture-library-ios#add-folly-and-other-dependencies) so that it's RN's responsibility to specify correct dependencies.